### PR TITLE
Ftr 1012 environment dialog ui

### LINKED ISSUE #4039: "Mobile first" for the new environment dialog
Make the new dialog mobile first

### PR DESCRIPTION
**Pull request by avukkonen**
closes #4039 
closes 1012  (already closed)

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx [hunk 1]
```diff
@@ -260,37 +247,57 @@ class NewEditAnnouncement extends SessionStateComponent<NewEditAnnouncementProps
...
+  
   render(){
     let content = (closeDialog: ()=>any) => [
       //FOR DESIGN CHECK https://github.com/Hacker0x01/react-datepicker
       (<div className="container container--new-announcement-options" key="1">
-         <DatePicker selected={this.state.startDate} onChange={this.handleDateChange.bind(this, "startDate")}
-           locale={this.props.i18n.time.getLocale()}/>
-         <div className="text text--announcer-times-decor">-</div>                   
-         <DatePicker selected={this.state.endDate} onChange={this.handleDateChange.bind(this, "endDate")}
+        <div className="environment-dialog__form-element--wrapper">
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx]: Where is the BEM block element, it seems that there is only  BEM element "environment-dialog__form-element--wrapper" directly here. 

Also if we use modifiers we need to include the base element classname too.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx [hunk 2]
```diff
@@ -260,37 +247,57 @@ class NewEditAnnouncement extends SessionStateComponent<NewEditAnnouncementProps
...
-      (<input key="3" type="text" className="container container--new-announcement-topic form-field form-field--new-announcement-topic"
-        placeholder={this.props.i18n.text.get('plugin.announcer.editannouncement.title.label')}
-        value={this.state.subject} onChange={this.onSubjectChange} autoFocus={!!this.props.announcement}/>),
-      (<CKEditor key="4" width="100%" height="grow" configuration={ckEditorConfig} extraPlugins={extraPlugins}
-       onChange={this.onCKEditorChange}>{this.state.text}</CKEditor>)
-    ]
-      
+      (
+      <div className="container container--new-announcement-title">    
+       <div className="environment-dialog__form-element--wrapper">
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx]: Where is the BEM block element, it seems that there is only BEM element "environment-dialog__form-element--wrapper" directly here.

Also if we use modifiers we need to include the base element classname too.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx [hunk 3]
```diff
@@ -260,37 +247,57 @@ class NewEditAnnouncement extends SessionStateComponent<NewEditAnnouncementProps
...
+       <div className="environment-dialog__form-element--wrapper">  
+         <div className="environment-dialog__form-element-label">{this.props.i18n.text.get('plugin.announcer.createannouncement.title.label')}</div>          
+         <input key="3" type="text" className="environment-dialog__input--title"          
+          value={this.state.subject} onChange={this.onSubjectChange} autoFocus={!!this.props.announcement}/>
+       </div>   
+      </div>
+      ),
+      (
+      <div className="container container--announcer-content">    
+        <div className="environment-dialog__form-element--wrapper">
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx]: Where is the BEM block element, it seems that there is only BEM element "environment-dialog__form-element--wrapper" directly here.

Also if we use modifiers we need to include the base element classname too.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/delete-announcement-dialog.tsx [hunk 4]
```diff
@@ -66,13 +66,13 @@ class DeleteAnnouncementDialog extends React.Component<DeleteAnnouncementDialogP
     let footer = (closeDialog: ()=>any)=>{
       return (          
          <div className="dialog__button-set">
-          <Link className="button button--cancel button--standard-cancel" onClick={closeDialog}>
-            {this.props.i18n.text.get('plugin.announcer.deleteDialog.cancelButton.label')}
-          </Link>
-          <Link className="button button--fatal button--standard-ok"
-            onClick={this.deleteAnnouncement.bind(this, closeDialog)} disabled={this.state.locked}>
+          <Link className="button button-dialog--execute"
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/delete-announcement-dialog.tsx]: This seems to be wrong regarding BEM. Button and button-dialog are both BEM blocks and it seems they are used as mixed blocks but button-dialog block has only modifier classname which is wrong.

Either use button or button-dialog only and then add modifier with base block name ie. "button-dialog button-dialog--execute".

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/delete-announcement-dialog.tsx [hunk 5]
```diff
@@ -66,13 +66,13 @@ class DeleteAnnouncementDialog extends React.Component<DeleteAnnouncementDialogP
...
+          onClick={this.deleteAnnouncement.bind(this, closeDialog)} disabled={this.state.locked}>
             {this.props.i18n.text.get('plugin.announcer.deleteDialog.deleteButton.label')}
           </Link>
+          <Link className="button button-dialog--cancel" onClick={closeDialog}>
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/delete-announcement-dialog.tsx]: This seems to be wrong regarding BEM. Button and button-dialog are both BEM blocks and it seems they are used as mixed blocks but button-dialog block has only modifier classname which is wrong.

Either use button or button-dialog only and then add modifier with base block name ie. "button-dialog button-dialog--cancel".

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx [hunk 6]
```diff
@@ -260,37 +247,57 @@ class NewEditAnnouncement extends SessionStateComponent<NewEditAnnouncementProps
...
-          </Button> : null}
-          <Button buttonModifiers={["warn", "standard-cancel"]} onClick={closeDialog} disabled={this.state.locked}>
-            {this.props.i18n.text.get('plugin.announcer.createannouncement.button.cancel')}
-          </Button>
-          <Button buttonModifiers="standard-ok" onClick={this.createOrModifyAnnouncement.bind(this, closeDialog)}>
+         <div className="environment-dialog__button-container">
+          <Button className="button button-dialog--execute" onClick={this.createOrModifyAnnouncement.bind(this, closeDialog)}>
             {this.props.i18n.text.get('plugin.announcer.createannouncement.button.send')}
           </Button> 
+          <Button className="button button-dialog--cancel" onClick={closeDialog} disabled={this.state.locked}>
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx]: This seems to be wrong regarding BEM. Button and button-dialog are both BEM blocks and it seems they are used as mixed blocks but button-dialog block has only modifier classname which is wrong.

Either use button or button-dialog only and then add modifier with base block name ie. "button-dialog button-dialog--cancel".

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx [hunk 7]
```diff
@@ -260,37 +247,57 @@ class NewEditAnnouncement extends SessionStateComponent<NewEditAnnouncementProps
...
+            {this.props.i18n.text.get('plugin.announcer.createannouncement.button.cancel')}
+          </Button>            
+          {this.recovered ? <Button className="button button-dialog--clear" onClick={this.clearUp} disabled={this.state.locked}>
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx]: This seems to be wrong regarding BEM. Button and button-dialog are both BEM blocks and it seems they are used as mixed blocks but button-dialog block has only modifier classname which is wrong.

Either use button or button-dialog only and then add modifier with base block name ie. "button-dialog button-dialog--execute".

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx [hunk 8]
```diff
@@ -260,37 +247,57 @@ class NewEditAnnouncement extends SessionStateComponent<NewEditAnnouncementProps
...
-           locale={this.props.i18n.time.getLocale()}/>
-         <div className="text text--announcer-times-decor">-</div>                   
-         <DatePicker selected={this.state.endDate} onChange={this.handleDateChange.bind(this, "endDate")}
+        <div className="environment-dialog__form-element--wrapper">  
+           <div className="environment-dialog__form-element-label">{this.props.i18n.text.get('plugin.announcer.createannouncement.startdate.label')}</div>          
+             <DatePicker selected={this.state.startDate} onChange={this.handleDateChange.bind(this, "startDate")}
+             locale={this.props.i18n.time.getLocale()}/>
+         </div>
+         <div className="text text--announcer-times-decor">-</div>
+         <div className="environment-dialog__form-element--wrapper">
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx]: If we use modifiers we need to include the base element class name too as modifier is an extra class name and dit cannot be the only class name of the html element.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx [hunk 9]
```diff
@@ -260,37 +247,57 @@ class NewEditAnnouncement extends SessionStateComponent<NewEditAnnouncementProps
...
-        value={this.state.subject} onChange={this.onSubjectChange} autoFocus={!!this.props.announcement}/>),
-      (<CKEditor key="4" width="100%" height="grow" configuration={ckEditorConfig} extraPlugins={extraPlugins}
-       onChange={this.onCKEditorChange}>{this.state.text}</CKEditor>)
-    ]
-      
+      (
+      <div className="container container--new-announcement-title">    
+       <div className="environment-dialog__form-element--wrapper">  
+         <div className="environment-dialog__form-element-label">{this.props.i18n.text.get('plugin.announcer.createannouncement.title.label')}</div>          
+         <input key="3" type="text" className="environment-dialog__input--title"
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx]: If we use modifiers we need to include the base element class name too as modifier is an extra class name and dit cannot be the only class name of the html element.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx [hunk 10]
```diff
@@ -260,37 +247,57 @@ class NewEditAnnouncement extends SessionStateComponent<NewEditAnnouncementProps
...
-         <div className="jumbo-dialog__button-container">
-          {this.recovered ? <Button buttonModifiers="danger" onClick={this.clearUp} disabled={this.state.locked}>
-            {this.props.i18n.text.get('clear draft')}
-          </Button> : null}
-          <Button buttonModifiers={["warn", "standard-cancel"]} onClick={closeDialog} disabled={this.state.locked}>
-            {this.props.i18n.text.get('plugin.announcer.createannouncement.button.cancel')}
-          </Button>
-          <Button buttonModifiers="standard-ok" onClick={this.createOrModifyAnnouncement.bind(this, closeDialog)}>
+         <div className="environment-dialog__button-container">
+          <Button className="button button-dialog--execute" onClick={this.createOrModifyAnnouncement.bind(this, closeDialog)}>
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application/new-edit-announcement.tsx]: This seems to be wrong regarding BEM. Button and button-dialog are both BEM blocks and it seems they are used as mixed blocks, however button-dialog block has only modifier class name which is wrong.

If we use modifiers we need to include the base block class name too as modifier is an extra class name and it cannot be the only class name of the html element.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application/new-message.tsx [hunk 11]
```diff
@@ -137,28 +138,40 @@ class CommunicatorNewMessage extends SessionStateComponent<CommunicatorNewMessag
...
       (<InputContactsAutofill modifier="new-messsage" key="1" hasGroupPermission placeholder={this.props.i18n.text.get('plugin.communicator.createmessage.title.recipients')}
         selectedItems={this.state.selectedItems} onChange={this.setSelectedItems} autofocus={!this.props.initialSelectedItems}></InputContactsAutofill>),
-      (<input key="2" type="text" className="form-field form-field--communicator-new-message-subject"
-        placeholder={this.props.i18n.text.get('plugin.communicator.createmessage.title.subject')}
-        value={this.state.subject} onChange={this.onSubjectChange} autoFocus={!!this.props.initialSelectedItems}/>),
-      (<CKEditor key="3" width="100%" height="grow" configuration={ckEditorConfig} extraPlugins={extraPlugins}
-       onChange={this.onCKEditorChange}>{this.state.text}</CKEditor>),
+      (
+       <div className="container container--communicator-subject">
+        <div className="environment-dialog__form-element--wrapper">
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application/new-message.tsx]: Where is the BEM block element, it seems that there is only BEM element "environment-dialog__form-element--wrapper" directly here.

If we use modifiers we need to include the base element class name too as modifier is an extra class name and it cannot be the only class name of the html element.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application/new-message.tsx [hunk 12]
```diff
@@ -137,28 +138,40 @@ class CommunicatorNewMessage extends SessionStateComponent<CommunicatorNewMessag
...
+        <div className="environment-dialog__form-element--wrapper">  
+          <div className="environment-dialog__form-element-label">{this.props.i18n.text.get('plugin.communicator.createmessage.title.subject')}</div>
+          <input key="2" type="text" className="environment-dialog__input--title"         
+          value={this.state.subject} onChange={this.onSubjectChange} autoFocus={!!this.props.initialSelectedItems}/>
+        </div> 
+        </div>
+        ),
+      (
+      <div className="container container--communicator-content">     
+        <div className="environment-dialog__form-element--wrapper">
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application/new-message.tsx]: Where is the BEM block element, it seems that there is only BEM element "environment-dialog__form-element--wrapper" directly here.

If we use modifiers we need to include the base element class name too as modifier is an extra class name and it cannot be the only class name of the html element.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/modify-area-dialog.tsx [hunk 13]
```diff
@@ -98,31 +98,37 @@ class DiscussionModifyArea extends SessionStateComponent<DiscussionModifyAreaPro
...
+          value={this.state.name} onChange={this.onNameChange} autoFocus/>
+        </div>
       </div>    
       
       ),
 
       (
        <div className="container container--new-discussion-area-description">          
-         <textarea key="2" className="form-field form-field--new-discussion-area-description"
+         <div className="environment-dialog__form-element--wrapper">
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/modify-area-dialog.tsx]: If we use modifiers we need to include the base element class name too as modifier is an extra class name and it cannot be the only class name of the html element.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/modify-area-dialog.tsx [hunk 14]
```diff
@@ -98,31 +98,37 @@ class DiscussionModifyArea extends SessionStateComponent<DiscussionModifyAreaPro
...
        
     let footer = (closeDialog: ()=>any)=>{
       return (          
-         <div className="jumbo-dialog__button-container">
-          {this.recovered ? <Button buttonModifiers="danger" onClick={this.clearUp} disabled={this.state.locked}>
-            {this.props.i18n.text.get('clear draft')}
-          </Button> : null}
-          <Button buttonModifiers={["warn", "standard-cancel"]} onClick={closeDialog} disabled={this.state.locked}>
+         <div className="environment-dialog__button-container">
+          <Button className="button button-dialog--clear" onClick={this.clearUp} disabled={this.state.locked}>
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/modify-area-dialog.tsx]: This seems to be wrong regarding BEM. Button and button-dialog are both BEM blocks and it seems they are used as mixed blocks, however button-dialog block has only modifier class name which is wrong.

If we use modifiers we need to include the base block class name too as modifier is an extra class name and it cannot be the only class name of the html element.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/modify-area-dialog.tsx [hunk 15]
```diff
@@ -98,31 +98,37 @@ class DiscussionModifyArea extends SessionStateComponent<DiscussionModifyAreaPro
...
+            {this.props.i18n.text.get('plugin.discussion.createarea.clearDraft')}
+          </Button>
+          <Button className="button button-dialog--cancel" onClick={closeDialog}>
             {this.props.i18n.text.get('plugin.discussion.createarea.cancel')}
           </Button>
-          <Button buttonModifiers="standard-ok" onClick={this.modifyArea.bind(this, closeDialog)} disabled={this.state.locked}>
+          <Button className="button button-dialog--execute" onClick={this.modifyArea.bind(this, closeDialog)} disabled={this.state.locked}>
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/modify-area-dialog.tsx]: This seems to be wrong regarding BEM. Button and button-dialog are both BEM blocks and it seems they are used as mixed blocks, however button-dialog block has only modifier class name which is wrong.

If we use modifiers we need to include the base block class name too as modifier is an extra class name and it cannot be the only class name of the html element.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/modify-area-dialog.tsx [hunk 16]
```diff
@@ -98,31 +98,37 @@ class DiscussionModifyArea extends SessionStateComponent<DiscussionModifyAreaPro
...
-         <div className="jumbo-dialog__button-container">
-          {this.recovered ? <Button buttonModifiers="danger" onClick={this.clearUp} disabled={this.state.locked}>
-            {this.props.i18n.text.get('clear draft')}
-          </Button> : null}
-          <Button buttonModifiers={["warn", "standard-cancel"]} onClick={closeDialog} disabled={this.state.locked}>
+         <div className="environment-dialog__button-container">
+          <Button className="button button-dialog--clear" onClick={this.clearUp} disabled={this.state.locked}>
+            {this.props.i18n.text.get('plugin.discussion.createarea.clearDraft')}
+          </Button>
+          <Button className="button button-dialog--cancel" onClick={closeDialog}>
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/modify-area-dialog.tsx]: This seems to be wrong regarding BEM. Button and button-dialog are both BEM blocks and it seems they are used as mixed blocks, however button-dialog block has only modifier class name which is wrong.

If we use modifiers we need to include the base block class name too as modifier is an extra class name and it cannot be the only class name of the html element.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/environment-dialog.scss [hunk 17]
```diff
@@ -0,0 +1,450 @@
...
+  outline: none;
+  padding: 3px 10px;
+
+  &:focus {
+    border: 0;
+    box-shadow: none;
+  }
+}
+
+.environment-dialog__form-element-tag-input--tag {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/environment-dialog.scss]: https://github.com/otavanopisto/muikku/issues/4036

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/environment-dialog.scss [hunk 18]
```diff
@@ -0,0 +1,433 @@
...
+  margin-bottom: 5px;
+}
+
+.environment-dialog__form-element-label-name {
+  display: block;
+  margin-bottom: 24px;
+  text-align: center;
+}
+
+.environment-dialog__form-element--toolbar-selector {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/environment-dialog.scss]: No.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/environment-dialog.scss [hunk 19]
```diff
@@ -0,0 +1,433 @@
...
+  padding: 3px 10px;
+  resize: none;
+  transition: box-shadow 0.3, border 0.3;
+}
+
+input:not([type="checkbox"]).environment-dialog__form-element, .environment-dialog__form-element input, textarea.environment-dialog__form-element, .environment-dialog__form-element textarea {
+  height: 34px;
+}
+
+input[type="checkbox"].environment-dialog__form-element {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/environment-dialog.scss]: No.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/environment-dialog.scss [hunk 20]
```diff
@@ -0,0 +1,433 @@
...
+  border: solid 2px $color-environment-dialog-form-element-border;
+  border-radius: 3px;
+  font-size: 0.875em;
+  outline: none;
+  padding: 3px 10px;
+  resize: none;
+  transition: box-shadow 0.3, border 0.3;
+}
+
+input:not([type="checkbox"]).environment-dialog__form-element, .environment-dialog__form-element input, textarea.environment-dialog__form-element, .environment-dialog__form-element textarea {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/environment-dialog.scss]: No.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+
+.container--message-actions {
+  padding: 10px 0;    
+}
+
+.container--discussion-profile-image,
+.container--guider-profile-image {
+  align-items: center;
+  border-radius: 100%;
+  color: #fff;
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss]: Garbage: this is fixed, but hound does not compute

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss [hunk 22]
```diff
@@ -0,0 +1,147 @@
...
+.container--communicator-message-links {
+  padding: 10px;
+}
+
+.container--new-discussion-options {
+  display: flex;
+}
+
+.container--new-discussion-thread-states {
+  background-color: #fff;
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss]: Garbage: this is fixed, but hound does not compute

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss [hunk 23]
```diff
@@ -0,0 +1,147 @@
...
+  margin-right: 5px;
+  width: 35px;
+}
+  
+.container--muikku-logo {
+ align-items: center;
+ display: flex;
+ flex-direction: row;
+ flex-flow: nowrap;
+ justify-content: center;
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss]: Garbage: this is fixed, but hound does not compute

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss [hunk 24]
```diff
@@ -0,0 +1,147 @@
...
+  height: 35px;
+  margin-right: 5px;
+  width: 35px;
+}
+  
+.container--muikku-logo {
+ align-items: center;
+ display: flex;
+ flex-direction: row;
+ flex-flow: nowrap;
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss]: Garbage: this is fixed, but hound does not compute

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss [hunk 25]
```diff
@@ -0,0 +1,147 @@
...
+  border-radius: 100%;
+  height: 35px;
+  margin-right: 5px;
+  width: 35px;
+}
+  
+.container--muikku-logo {
+ align-items: center;
+ display: flex;
+ flex-direction: row;
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss]: Garbage: this is fixed, but hound does not compute

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss [hunk 26]
```diff
@@ -0,0 +1,147 @@
...
+.container--profile-image-in-menu {
+  border-radius: 100%;
+  height: 35px;
+  margin-right: 5px;
+  width: 35px;
+}
+  
+.container--muikku-logo {
+ align-items: center;
+ display: flex;
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss]: Garbage: this is fixed, but hound does not compute

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss [hunk 27]
```diff
@@ -0,0 +1,147 @@
...
+  @include breakpoint($breakpoint-pad) {
+    flex-direction: column;
+  }
+}
+  
+.container--frontpage-section {
+  margin-top: 1.5rem;
+  padding: 0 $container-mobile-gutter;
+
+  @include breakpoint($breakpoint-pad){
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/container.scss]: Garbage: this is fixed, but hound does not compute

**Pull request merged**